src/main/java/com/iota/iri/controllers/TransactionViewModel.java

```diff
@@ -160,11 +160,11 @@ public void delete(Tangle tangle) throws Exception {
 
     public List<Pair<Indexable, Persistable>> getMetadataSaveBatch() throws Exception {
         List<Pair<Indexable, Persistable>> hashesList = new ArrayList<>();
-        hashesList.add(new Pair<>(getAddressHash(), new Address(hash)));
-        hashesList.add(new Pair<>(getBundleHash(), new Bundle(hash)));
```

Comment:
also, there is an implicit use of this `getMetadataSaveBatch` function and that is to fill the lazy fields in the transactionVM - this PR fails when storing non-tail transactions b/c of `bundle` bytes not being present:
```java.lang.NullPointerException: null
	at com.iota.iri.model.Transaction.metadata(Transaction.java:65) ~[classes/:na]
	at com.iota.iri.storage.rocksDB.RocksDBPersistenceProvider.saveBatch(RocksDBPersistenceProvider.java:346) ~[classes/:na]
	at com.iota.iri.storage.Tangle.saveBatch(Tangle.java:53) ~[classes/:na]
	at com.iota.iri.controllers.TransactionViewModel.store(TransactionViewModel.java:207) ~[classes/:na]
	at com.iota.iri.network.Node.processReceivedData(Node.java:376) [classes/:na]
	at com.iota.iri.network.Node.processReceivedDataFromQueue(Node.java:358) [classes/:na]
	at com.iota.iri.network.Node.lambda$spawnProcessReceivedThread$8(Node.java:559) [classes/:na]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[na:1.8.0_151]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[na:1.8.0_151]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[na:1.8.0_151]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[na:1.8.0_151]
	at java.lang.Thread.run(Thread.java:748) ~[na:1.8.0_151]```

Reply:
tested reverting the changes to `TransactionViewModel.java` - works well.